src/main/java/com/matt/forgehax/mods/DerpMod.java

```diff
@@ -0,0 +1,244 @@
+          .changed(change -> MC.addScheduledTask(() -> {
+            if (!isEnabled()) return;
+            if (change.getFrom() && !change.getTo()) sneakDisabled();
+            if (!change.getFrom() && change.getTo()) {
+              EntityPlayer player = getLocalPlayer();
+              sneaking = player != null && player.isSneaking();
+            }
+          }))
+          .defaultTo(false)
+          .build();
+
+  public final Setting<Float> sneakChance =
+      getCommandStub()
+          .builders()
+          .<Float>newSettingBuilder()
+          .name("sneak_chance")
+          .description("Chance to sneak for every derp")
+          .min(0f)
+          .max(1f)
+          .defaultTo(1f)
+          .build();
+
+  public final Setting<Boolean> hit =
+      getCommandStub()
+          .builders()
+          .<Boolean>newSettingBuilder()
+          .name("hit")
+          .description("Randomly hit every derp")
+          .defaultTo(true)
+          .build();
+
+  public final Setting<Float> hitChance =
+      getCommandStub()
+          .builders()
+          .<Float>newSettingBuilder()
+          .name("hit_chance")
+          .description("Chance to hit for every derp")
+          .min(0f)
+          .max(1f)
+          .defaultTo(.25f)
+          .build();
+
+  public final Setting<Float> hitPan =
+      getCommandStub()
+          .builders()
+          .<Float>newSettingBuilder()
+          .name("hit_pan")
+          .description("Chance for any hit to be the off hand")
+          .min(0f)
+          .max(1f)
+          .defaultTo(.5f)
+          .build();
+
+  private void sendRotatePacket(EntityPlayer player, float pitch, float yaw) {
+    ignoreAndSend(new CPacketPlayer.Rotation(yaw, pitch, player.onGround));
+  }
+
+  private void rotateDisabled() {
+    EntityPlayer player = getLocalPlayer();
+    if (player == null) return;
+
+    sendRotatePacket(player, player.rotationPitch, player.rotationYawHead);
+  }
+
+  private CPacketEntityAction.Action actionFromSneaking(boolean sneaking) {
+    return sneaking
+        ? CPacketEntityAction.Action.START_SNEAKING
+        : CPacketEntityAction.Action.STOP_SNEAKING;
+  }
+
+  private void sendSneakPacket(EntityPlayer player, boolean sneaking) {
+    this.sneaking = sneaking;
+    ignoreAndSend(new CPacketEntityAction(player, actionFromSneaking(sneaking)));
+  }
+
+  private void sneakDisabled() {
+    EntityPlayer player = getLocalPlayer();
+    if (player == null) return;
+
+    sendSneakPacket(player, player.isSneaking());
+  }
+
+  @Override
+  protected void onDisabled() {
+    if (rotate.get()) rotateDisabled();
+    if (sneak.get()) sneakDisabled();
+  }
+
+  private void sendHitPacket(EnumHand hand) {
+    ignoreAndSend(new CPacketAnimation(hand));
+  }
+
+  private boolean chance(float chance) {
+    return chance == 1f || (chance != 0f && Math.random() <= chance);
+  }
+
+  @SubscribeEvent
+  public void onLocalPlayerUpdate(LocalPlayerUpdateEvent event) {
```

Comment:
I have a service dedicated to handling the players movements and rotations. Look at Aimbot, Autoplace, Nuker, or Scaffold for an example on how to use it.

Reply:
Okay, I'll put that into the newest commit.

Reply:
What do you want me to do with it?

Reply:
It works by prioritizing each mod that uses it. If two mods try and change the players view at the same time, only the one with the highest priority will be allowed to do it. The mod that gets top priority then has its executor with the exception that the player is looking close enough to its target angle.

Reply:
Derp shouldn't change the view. Derp just injects packets to make the server _think_ that. Other mods will work fine, and if the anticheat gets triggered you can disable Derp.

Reply:
Yes. That is what the rotation manager handles. You can only send so many PositionRotation packets before the server flags you. And since most servers use anti cheats, new ForgeHax mods should be designed to work with them.

Reply:
> You can only send so many PositionRotation packets before the server flags you

1. This mod intercepts PositionRotation packets and replaces them with Position packets
2. This mod has a speed setting so it doesn't send too many packets anyway
3. This mod has timer sync so it still doesn't accidentally send too many packets with timer on
4. This mod does work with anticheats to some extent. I can still move around, mine blocks, etc. on servers with anticheats. I have tested this. With sneak on you get rubberbanded sometimes though.

Reply:
`sendRotatePacket(player, rng.nextInt(180) - 90, rng.nextInt(360) - 180);`

this line right here sends an additional rotation packet. Sending even 1 packet is too much. On top of that, if you change your view angle too far from your last view angle anti-cheats will flag it and clamp it or just drop it. Why bother with intercepting packets, timer sync, and speed control, when the rotation manager handles all that for you? On top of that, it will allow other mods to function with derp.

Reply:
RotateManager doesn't send packets as much as I'd like it to. It also seems to be biased towards where the player's currently looking. I think I'm going to stay entirely packet-based.

> this line right here sends an additional rotation packet. Sending even 1 packet is too much.

Then turn off Derp.

> On top of that, if you change your view angle too far from your last view angle anti-cheats will flag it and clamp it or just drop it.

Then turn off rotate.

> Why bother with intercepting packets, timer sync, and speed control, when the rotation manager handles all that for you?

Because the rotation manager doesn't fit with the mod.

> On top of that, it will allow other mods to function with derp.

Give me an example of a mod that doesn't work with derp. I haven't found a single one.

Reply:
Just like my other mods, Derp works the way it does because that's how it's supposed to work. It doesn't bypass anticheats because anticheats will flag a player derping no matter how much you try to fool it. It's configurable, it's invisible to the player using it, it's everything a high-quality derp mod should be. If you believe ForgeHax does not need a derp mod, then it will stay in my custom fork. ¯\\\_(ツ)\_/¯

Reply:
Sorry, when I design anything I have anticheats in mind. So when I say compatible with other mods, I mean it wont cause anticheats to flag you because 2 mods are trying to hijack the players view angles. The majority of ForgeHax users play on 2b2t and its clones, all of which use anticheats. I don't want to merge a mod that will just get its users, who might have just spent hours waiting through a queue, kicked because a mod doesn't work around the anti cheat.

Reply:
I hate to put this ultimatum on the mod, but hear me out. Most users play on servers with anticheats, what is the point of this mod if the majority of people using it will be unable to?

Reply:
The defaults for this mod send about 20-40 packets per second. I believe I could crank this up to timer 5 or so on a vanilla server before I got kicked for sending too many packets. Then I added timer sync, which is meant to keep the packets sent down to 20-40 per second, even with timer on. Now, this is all assuming onLocalPlayerUpdate is called once per tick and always exactly once per tick. Is that assumption correct?

I understand what you're trying to say here, and I'm having a bit of trouble trying to phrase what I'm trying to say. A derp mod by design sends many many packets, users should know this, and keep it off, keep it at sane speeds, or only toggle it on for a second or two. This one sends enough packets to get you kicked at speed 5 or so on vanilla. I'm not sure about anticheats.

In fact, let me go onto 0b0t and 8b8t and see how far I can crank this before getting kicked. I'll get back to you.

Reply:
Try it on constantiam.net,

Reply:
That's on my list too.

Reply:
0b0t: kick at timer 5 with timer-sync turned off
8b8t: No kick even at timer 10
Constantiam: No kick even at timer 12

Options:
Rotate is on with chance 1
Sneak is on with chance 1
Hit is on with chance 0.25 (the default)

I could wait 8 hours to get into 2b2t, but, nah.

Reply:
The real question is can you move with it on. NCP only suppresses too many packets.

Reply:
Yes, yes you can. Even on non-anarchy servers. I even had sneak on and I could move fine.

Reply:
On Constantiam I was able to move around fine with derp and timer 12